Having trouble importing deepmerge with Webpack

When importing deepmerge with `import * as merge from 'deepmerge';` throws `Error: merge is not a function` in the app, while the tests run just fine.

When importing with `import merge from `deepmerge;`, the app works fine, but the jest tests throw `TypeError: deepmerge_1.default is not a function`.

I have temporarily patched the issue in my code with an `if/else` block to reassign the correct function to the variable, but I feel there must be a better way.

Maybe jest is misconfigured. Has anyone ever encountered a similar issue or can provide a configuration template that is proven to be working?

This describes a similar issue to what I am experiencing https://github.com/aurelia/skeleton-navigation/issues/606, though I am not using Aurelia, nor Moment. It just seems like the same _kind_ of problem.

--------

**Node version**: 8.6.0
**NPM version**: 5.5.1
**Webpack version**: 3.6.0
**Typescript version**: 2.5.3
**deepmerge version**: 2.0.1

-------

tsconfig.json

```javascript
{
  "compilerOptions": {
    "outDir": "./dist/",
    "sourceMap": true,
    "noImplicitAny": false,
    "module": "commonjs",
    "target": "es5",
    "lib": ["es7", "es2017", "dom"],
    "jsx": "preserve",
    "allowJs": true
  },
  "include": ["src/**/*"],
  "exclude": ["node_modules", "**/*.test.ts"]
}
```

-------

.babelrc

```javascript
{
  "presets": ["env", "react"],
  "plugins": [
    "react-css-modules"
  ]
}
```

-------

jest.config.json

```javascript
{
  "transform": {
    ".(ts|tsx)": "<rootDir>/node_modules/ts-jest/preprocessor.js"
  },
  "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(ts|tsx|js)$",
  "moduleFileExtensions": ["ts", "tsx", "js", "jsx", "json"]
}
```

I have a similar problem where it's looking in all places, just not the node_modules:
```bash
ERROR in /Users/-----/.ghq/github.com/mesqueeb/VuexEasyFirestore/src/utils/setDefaultValues.js
Module not found: Error: Can't resolve 'deepmerge' in '/Users/-----/.ghq/github.com/mesqueeb/VuexEasyFirestore/src/utils'
 @ /Users/-----/.ghq/github.com/mesqueeb/VuexEasyFirestore/src/utils/setDefaultValues.js 1:0-29 11:9-14
 @ /Users/-----/.ghq/github.com/mesqueeb/VuexEasyFirestore/src/module/actions.js
 @ /Users/-----/.ghq/github.com/mesqueeb/VuexEasyFirestore/src/module/index.js
 @ /Users/-----/.ghq/github.com/mesqueeb/VuexEasyFirestore/src/index.js
 @ ./src/store/store.js
 @ ./src/store/index.js
```

--- a/changelog.md
+++ b/changelog.md
@@ -1,3 +1,7 @@
+# [3.0.0](https://github.com/KyleAMathews/deepmerge/releases/tag/v3.0.0)
+
+- drop ES module build [#123](https://github.com/KyleAMathews/deepmerge/issues/123)
+
 # [2.2.1](https://github.com/KyleAMathews/deepmerge/releases/tag/v2.2.1)
 
 - bug: typescript export type was wrong [#121](https://github.com/KyleAMathews/deepmerge/pull/121)

--- a/package.json
+++ b/package.json
@@ -17,7 +17,6 @@
     "url": "git://github.com/KyleAMathews/deepmerge.git"
   },
   "main": "dist/umd.js",
-  "module": "dist/es.js",
   "engines": {
     "node": ">=0.10.0"
   },

--- a/readme.md
+++ b/readme.md
@@ -80,7 +80,7 @@ var merge = require('deepmerge')
 
 ES Modules:
 ```
-import * from 'deepmerge'
+import * as deepmerge from 'deepmerge'
 ```
 
 (support for `import merge from 'deepmerge'` was removed because of a [Webpack bug](https://github.com/webpack/webpack/issues/6584)).

--- a/readme.md
+++ b/readme.md
@@ -5,24 +5,11 @@ Merges the enumerable attributes of two or more objects deeply.
 > UMD bundle is 567B minified+gzipped
 
 
-### Migration from 1.x to 2.0.0
+### Migration from 1.x to 2+
 
 [***Check out the changes from version 1.x to 2.0.0***](https://github.com/KyleAMathews/deepmerge/blob/master/changelog.md#200)
 
 For the legacy array element-merging algorithm, see [the `arrayMerge` option below](#arraymerge).
-
-
-### Webpack bug
-
-If you have `require('deepmerge')` (as opposed to `import merge from 'deepmerge'`) anywhere in your codebase, Webpack 3 and 4 have a bug that [breaks bundling](https://github.com/webpack/webpack/issues/6584).
-
-If you see `Error: merge is not a function`, add this alias to your Webpack config:
-
-```
-alias: {
-	deepmerge$: path.resolve(__dirname, 'node_modules/deepmerge/dist/umd.js'),
-}
-```
 
 
 ## Getting Started
@@ -93,8 +80,10 @@ var merge = require('deepmerge')
 
 ES Modules:
 ```
-import merge from 'deepmerge'
+import * from 'deepmerge'
 ```
+
+(support for `import merge from 'deepmerge'` was removed because of a [Webpack bug](https://github.com/webpack/webpack/issues/6584)).
 
 
 # API

--- a/rollup.config.js
+++ b/rollup.config.js
@@ -1,17 +1,16 @@
 import resolve from 'rollup-plugin-node-resolve'
 import commonjs from 'rollup-plugin-commonjs'
 
-var pkg = require('./package.json')
+const pkg = require(`./package.json`)
 
 export default {
-	input: 'index.js',
-	name: 'deepmerge',
+	input: `index.js`,
+	name: `deepmerge`,
 	plugins: [
 		commonjs(),
 		resolve(),
 	],
 	output: [
-		{ file: pkg.main, format: 'umd' },
-		{ file: pkg.module, format: 'es' },
+		{ file: pkg.main, format: `umd` },
 	],
 }

